src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -216,25 +216,33 @@ internal bool ShouldRunNullableWalker
             {
                 if (!_lazyShouldRunNullableWalker.HasValue())
                 {
+                    _lazyShouldRunNullableWalker = calculateResult();
+                }
+                return _lazyShouldRunNullableWalker.Value();
+
+                ThreeState calculateResult()
+                {
+                    var feature = Feature("nullableAnalysis");
+                    if (feature == "false")
```

Comment:
Consider having a central method that returns a `ThreeState` for the `"nullableAnalysis"` flag rather than having it interpreted in multiple places.